src/js/BookReader.js

```diff
@@ -1682,42 +1684,80 @@ BookReader.prototype._scrollAmount = function() {
   return parseInt(0.9 * this.refs.$brContainer.prop('clientHeight'));
 };
 
-BookReader.prototype.prefetchImg = function(index) {
-  var pageURI = this._getPageURI(index, this.reduce);
+/**
+ * Used by 2up
+ * Fetches the image for requested index & saves in `this.prefetchedImgs`
+ * Does not re-request if image is in the
+ *
+ * @param {Number} index
+ * @param {Boolean} fetchNow
+ *   - flag to allow for non-viewable page to be immediately requested
+ *     - this allows for "2up to prepare a page flip"
+ */
+BookReader.prototype.prefetchImg = async function(index, fetchNow = false) {
+  const pageURI = this._getPageURI(index, this.reduce);
   const pageURISrcset = this.options.useSrcSet ? this._getPageURISrcset(index, this.reduce) : [];
 
   // Load image if not loaded or URI has changed (e.g. due to scaling)
-  var loadImage = false;
+  let loadImage = false;
+  const wasPrefetchedSmaller = this.prefetchedImgs[index]?.reduce > this.reduce;
   if (undefined == this.prefetchedImgs[index]) {
     loadImage = true;
-  } else if (pageURI != this.prefetchedImgs[index].uri) {
+  } else if (wasPrefetchedSmaller) {
     loadImage = true;
+  } else if (!wasPrefetchedSmaller && (pageURI != this.prefetchedImgs[index]?.uri)) {
+    loadImage = true;
+  }
+
+  if (!loadImage) {
+    return;
   }
 
-  if (loadImage) {
-    const pageContainer = this._createPageContainer(index);
-    $('<img />', {
-      'class': 'BRpageimage',
-      'alt': 'Book page image',
-      src: pageURI,
-      srcset: pageURISrcset
-    }).appendTo(pageContainer);
-    if (index < 0 || index > (this._models.book.getNumLeafs() - 1) ) {
-      // Facing page at beginning or end, or beyond
-      pageContainer.addClass('BRemptypage');
+  if (wasPrefetchedSmaller) {
+    /* cancel the request if still pending */
+    $(this.prefetchedImgs[index]).find('img').attr('src', '');
+  }
+
+  const $pageContainer = this._createPageContainer(index);
+  const $imgShell = this._modes.mode2Up.createPageImgShell(index, pageURI, this.reduce, $pageContainer);
+
+  /** set uri in img tag to start request & save in `this.prefetchedImgs` */
+  const fetchImageAndRegister = () => {
+    const $imgEl = $($imgShell).find('img');
+    $imgEl.attr('src', pageURI);
+    if (pageURISrcset.length) {
+      $imgEl.attr('srcSet', pageURISrcset);
     }
-    pageContainer[0].uri = pageURI; // browser may rewrite src so we stash raw URI here
-    this.prefetchedImgs[index] = pageContainer[0];
+    $($imgEl).load(() => {
+      console.log('** PREFETCH DONE: index, this.reduce', index, this.reduce);
+    }).error(() => {
+      console.log('** PREFETCH ERROR: index, this.reduce', index, this.reduce);
```

Comment:
(saving room for error handling... - currently unsure what it is right now, so we can comment out if needed)